README.md

```diff
@@ -7,142 +9,92 @@
-- Commands operating on string values
-- Commands operating on hashes
-- Commands operating on lists
-- Commands operating on sets
-- Commands operating on sorted sets
-- Commands operating on streams
-- Transactions
-- Pipelining
-- Publish/Subscribe
-- Persistence control commands
-- Remote server control commands
-- Connection pooling
-- Sharding (MD5, MurmurHash)
-- Key-tags for sharding
-- Sharding with pipelining
-- Scripting with pipelining
-- Redis Cluster
+## Getting started
 
-## How do I use it?
-
-You can download the latest build at: 
-    http://github.com/redis/jedis/releases
-
-Or use it as a maven dependency:
-
-### Official Releases
+To get started with Jedis, first add it as a dependency in your Java project. If you're using Maven, that looks like this:
 
 ```xml
 <dependency>
     <groupId>redis.clients</groupId>
     <artifactId>jedis</artifactId>
-    <version>3.7.0</version>
+    <version>4.0.0</version>
 </dependency>
 ```
 
-### Snapshots
+Next, you'll need to connect to Redis. For many applications, it's best to use a connection pool. You can instantiate a Jedis connection pool like so:
 
-```xml
-  <repositories>
-    <repository>
-      <id>snapshots-repo</id>
-      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
-    </repository>
-  </repositories>
+```java
+JedisPool pool = new JedisPool("localhost", 6379);
 ```
 
-and
+Once you have a `JedisPool` instance, you can use a [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) block to get a connection and run Redis commands.
 
-```xml
-  <dependencies>
-    <dependency>
-      <groupId>redis.clients</groupId>
-      <artifactId>jedis</artifactId>
-      <version>4.0.0-SNAPSHOT</version>
-    </dependency>
-  </dependencies>
+Here's how to run a single [SET](https://redis.io/commands/set) command within a
+*try-with-resources* block:
+
+```java
+try (Jedis jedis = pool.getResource()) {
+  jedis.set("clientName", "Jedis");
+}
 ```
 
-or, for upcoming minor release
+`Jedis` instances implement most Redis commands. See the [Jedis Javadocs](https://www.javadoc.io/doc/redis.clients/jedis/latest/redis/clients/jedis/Jedis.html) for a complete list of supported commands.
 
-```xml
-  <dependencies>
-    <dependency>
-      <groupId>redis.clients</groupId>
-      <artifactId>jedis</artifactId>
-      <version>3.8.0-SNAPSHOT</version>
-    </dependency>
-  </dependencies>
+## Using RedisJSON and Redis modules
+
+Jedis provides support for some of the [Redis modules](https://redis.io/modules), most notably [RedisJSON](https://oss.redis.com/redisjson/) and [RediSearch](https://oss.redis.com/redisearch/).
+
+See the [RedisJSON Jedis Quick Start](docs/redisjson.md) for details.
+
+## Connecting to a Redis cluster
+
+Jedis lets you connect to Redis Clusters, supporting the [Redis Cluster Spec](http://redis.io/topics/cluster-spec). To do this, you'll need to connect using `JedisCluster`. See the example below:
```

Comment:
`[Redis Cluster Spec](http://redis.io/topics/cluster-spec)` should be `[Redis Cluster Specification](http://redis.io/topics/cluster-spec)`